File: core/src/main/java/com/bindglam/utility/compatibility/NexoCompatibility.java
package com.bindglam.utility.compatibility;

import com.nexomc.nexo.NexoPlugin;
import com.nexomc.nexo.api.NexoItems;
import com.nexomc.nexo.glyphs.Shift;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.Nullable;

import java.util.Objects;

public class NexoCompatibility implements Compatibility {
    @Override
    public @Nullable Component getGlyphOrNull(String id, int offsetX) {
        if(NexoPlugin.instance().fontManager().glyphFromID(id) == null)
            return null;

        return Component.text(Shift.INSTANCE.of(offsetX)).append(Objects.requireNonNull(NexoPlugin.instance().fontManager().glyphFromID(id)).glyphComponent(true)).color(NamedTextColor.WHITE);
    }

    @Override
    public @Nullable ItemStack getCustomItemOrNull(String id) {
        if(!NexoItems.exists(id))
            return null;

        return Objects.requireNonNull(NexoItems.itemFromId(id)).build();
    }

    @Override
    public @Nullable String getCustomItemIdByItemStack(@Nullable ItemStack itemStack) {
        return NexoItems.idFromItem(itemStack);
    }
}

File: core/src/main/kotlin/com/bindglam/utility/compatibility/NexoCompatibility.kt
package com.bindglam.utility.compatibility

import com.nexomc.nexo.NexoPlugin
import com.nexomc.nexo.api.NexoItems
import com.nexomc.nexo.glyphs.Shift
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.NamedTextColor
import org.bukkit.inventory.ItemStack

class NexoCompatibility : Compatibility {
    override fun getGlyph(id: String, offsetX: Int): Component {
        return Component.text(Shift.of(offsetX)).append(NexoPlugin.instance().fontManager().glyphFromID(id)!!.glyphComponent(true)).color(NamedTextColor.WHITE)
    }

    override fun getGlyphOrNull(id: String, offsetX: Int): Component? {
        if (NexoPlugin.instance().fontManager().glyphFromID(id) == null) return null
        return getGlyph(id, offsetX)
    }

    override fun getCustomItem(id: String): ItemStack {
        return NexoItems.itemFromId(id)!!.build()
    }

    override fun getCustomItemOrNull(id: String): ItemStack? {
        if (!NexoItems.exists(id)) return null
        return getCustomItem(id)
    }
}